Add Eternal Lands Game

pjbroad opened the pull request:
Please consider this packaging of Eternal Lands https://github.com/raduprv/Eternal-Lands.
Thanks,
Paul

pjbroad commented:
bot, build org.pjbroad.EternallandsClient

AsciiWolf commented:
bot, build org.pjbroad.EternallandsClient

AsciiWolf commented:
Build [failed](https://flathub.org/builds/#/builders/39/builds/318) on aarch64. Also, the `eternallands.png` icon seems to be too small, consider using a better icon, for example [this one](https://dashboard.snapcraft.io/site_media/appmedia/2018/07/gallery_4_9_63565.png) from the Snap package. (Although I am not sure what license this icon has, but it should be generated from [this svg](https://github.com/raduprv/Eternal-Lands/blob/master/pkgfiles/snap-package/eternallands.svg) that unfortunately seems to be broken in git.)

AsciiWolf commented on org.pjbroad.EternallandsClient.json:
```diff
@@ -0,0 +1,172 @@
+				{
+					"type": "file",
+					"path": "eternallands.png",
+					"dest-filename": "org.pjbroad.EternallandsClient.png"
```
By the way, you can rename the desktop/icon files directly in the git repo.

pjbroad commented:
@AsciiWolf thanks for looking at this package and for the helpful comments.  I am not seeing the build failure but I've updated the config scripts for the cal3d component.  I've also renamed the desktop and icon files and added a larger icon.

AsciiWolf commented:
No problem. :-) The [new build](https://flathub.org/builds/#/builders/2/builds/3542) passed on all archs without any issues.

pjbroad commented:
Good news, thanks.  I get a "502 bad gateway" with that link but I guess I have no need to check.  Is there anything else I need to change?  This is my first flatpak so I do not what to expect to happen next.

AsciiWolf commented:
> I get a "502 bad gateway" with that link

Weird, it's working fine for me.

> Is there anything else I need to change?

I didn't test the actual game yet, but it looks good to me.

pjbroad commented:
OK, thanks.  The link ok OK now.

AsciiWolf commented on org.pjbroad.EternallandsClient.json:
```diff
@@ -0,0 +1,169 @@
+		"--socket=pulseaudio",
+		"--share=network",
+		"--device=dri",
+		"--filesystem=~/.elc:create"
```
You can use `--persist=.elc`.

AsciiWolf replied:
(This means that the .elc directory will be created in `~/.var/app/org.pjbroad.EternallandsClient/` instead of `~/`.)

pjbroad replied:
I must admit that I did not fully understand the description in the docs about what --persist is for.  What I want is to be able to use ~/.elc if it exists and create ~/.elc if is does not already exist. Also, for changes in ~/.elc to be persistent and unaffected if the package is uninstalled.  The reason for is is to provide compatibility with existing installs of the game that do not use flatpak and to keep the "standard" location these user files the same so others can help easily with issues.  It looks like "--filesystem=~/.elc:create" and "--persist=~/.elc:create" both do this but I do not know which is best or what they do differently.  I'd appreciate your advise on this.

barthalion replied:
The directory is being created by `--persist=.elc` too if missing; none of the files created by applications are removed on uninstall. Note the path is already homedir relative.

It binds '~/.elc` in the sandbox to '~/.var/app/appid/.elc` so it won't migrate the already existing data.

pjbroad replied:
Thanks for the further explanation, but I still don't get it 100%, the need for mapping in particular.  I need the user files to be in the ~/.elc/ for all the reasons above.  Either already there or created there.  I've tried experimented with "--persist=\~/.elc" but that does not create the directory if missing "--persist=\~/.elc:create" does create it so I'll use that.  I'll remove the leading ~ but it looks more obvious what's happening with it present I'd suggest.

pjbroad replied:
So I've done a bunch more testing, including removing the files that get stored in ~/.var/app to make sure its all celan.  I was wrong, --persist does not work, ~/elc/ path is not created.  Therefore I'm sticking with --filesystem which does work.  Hope that's OK.

barthalion replied:
I'll try to explain it better (sorry if I fail!) The diference between --filesystem is that application sees it as is; so if you pass --filesystem=~/foo, it will be visible from app perspective as ~/foo and it will be stored outside the sandbox also as ~/foo. Persist is a hack to make applications not following XDG spec see their directory at correct location for them (so, `~/.eic` here; only in the sandbox!) but in reality stored elsewhere outside the sandbox, in this instance `~/.var/app/org.pjbroad.EternallandsClient/.eic`. This is why you don't see it created – it is being created from your not-in-sandbox point of view but elsewhere.

I'd much prefer if we could keep `--persist` but of course I can't force you if that's your decision.

pjbroad replied:
I think I finally get it, many thanks for taking more time to explain and I can see why that approach is useful / preferred.  However, for this application, user will likely interact with their config files stored in ~/.elc/, creating new player configs and changing elements of the game UI etc.  Existing players of the game switching to flatpak will want their existing config files to be used.  If they choose to switch back to some other packages version or one they have built themselves, they will want to keep using the same config files.  If we use --persist the user will have to manage this transition themselves.  Also, from a support perspective, using the same location across all Linux versions is desirable.  Hopefully I have understood now and have suitably explained my reasoning.

AsciiWolf commented on org.pjbroad.EternallandsClient.desktop:
```diff
@@ -0,0 +1,24 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Eternal Lands (flatpak)
```
I would remove the " (flatpak)" suffix. It's not important (and can be confusing) to the end-user.

pjbroad replied:
Yep, this has largely been for my own benefit during development as I'm also developing the snap version and have other installs present too.  I'll take it out before finishing if this pull request is successful :)

barthalion commented:
bot, build org.pjbroad.EternallandsClient

barthalion commented on org.pjbroad.EternallandsClient.json:
```diff
@@ -0,0 +1,169 @@
+		},
+
+		{
+			"name": "exports",
```
This doesn't have to be a separate module and can be part of `client`.

pjbroad replied:
I found it easier to make modifications keeping it separate to save rebuilds but I can move it.

barthalion replied:
Makes sense.

barthalion commented on org.pjbroad.EternallandsClient.json:
```diff
@@ -0,0 +1,169 @@
+		},
+
+		{
+			"name": "launcher",
```
This doesn't have to be a separate module and can be part of `client`.

barthalion commented on org.pjbroad.EternallandsClient.json:
```diff
@@ -0,0 +1,169 @@
+			"name": "client",
+			"buildsystem": "simple",
+			"build-commands": [
+					"make -f Makefile.linux release PLATFORM=\"\" -j 4",
```
Use `$FLATPAK_BUILDER_N_JOBS` instead of `4`.

barthalion commented on client-patch:
```diff
@@ -0,0 +1,40 @@
+diff --git a/Makefile.linux b/Makefile.linux
```
Shouldn't it be part of repo with source code?

pjbroad replied:
Yes, but in a future release.  I'm creating this flatpak based on the stable release as advised.

barthalion commented on launcher.sh:
```diff
@@ -0,0 +1,44 @@
+#! /bin/sh
```
Shouldn't it be part of repo with source code?

pjbroad replied:
Yes, but as before in a future release.

barthalion commented on org.pjbroad.EternallandsClient.appdata.xml:
```diff
@@ -0,0 +1,46 @@
+		</screenshot>
+	</screenshots>
+	<metadata_license>CC0-1.0</metadata_license>
+	<project_license>QPL-1.0 LicenseRef-Propritary</project_license>
```
AFAIK there should be `and` between two licenses.

AsciiWolf replied:
By the way, I just noticed a typo: *LicenseRef-Propritary* should be probably *LicenseRef-proprietary*. ;-)

pjbroad replied:
OK, thanks.

barthalion commented on org.pjbroad.EternallandsClient.appdata.xml:
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
It would be better to maintain this file in main repo so everyone can benefit from it.

barthalion replied:
Also the same applies to the png icon.

pjbroad replied:
Yes, but as before in a future release

barthalion commented on launcher.sh:
```diff
@@ -0,0 +1,44 @@
+userdir=~/.elc
+serverfile=servers.lst
+inifile=el.ini
+exename=/app/bin/el.x86.linux.bin
```
It doesn't work with other (arm) architectures?

pjbroad replied:
Yes it does but I agree the current name is confusing.  I'll modify it in the make file patch for now.

pjbroad commented:
Thanks for the feedback.  I've made most of the changes suggested and commented on all of them.  Hopefully you both happy with what I've said / done.

barthalion commented:
bot, build org.pjbroad.EternallandsClient

barthalion commented:
Repository has been created: https://github.com/flathub/org.pjbroad.EternallandsClient

Thanks!
